pkg/reconciler/events/auditlogs/auditlogs.go

```diff
@@ -263,32 +264,36 @@ func (c *Reconciler) removeFinalizer(s *v1alpha1.CloudAuditLogsSource) {
 	s.Finalizers = finalizers.List()
 }
 
-func (c *Reconciler) updateStatus(ctx context.Context, desired *v1alpha1.CloudAuditLogsSource) (*v1alpha1.CloudAuditLogsSource, error) {
-	source, err := c.auditLogsSourceLister.CloudAuditLogsSources(desired.Namespace).Get(desired.Name)
-	if err != nil {
-		return nil, err
-	}
-	// Check if there is anything to update.
-	if equality.Semantic.DeepEqual(source.Status, desired.Status) {
-		return source, nil
-	}
-	becomesReady := desired.Status.IsReady() && !source.Status.IsReady()
-
-	// Don't modify the informers copy.
-	existing := source.DeepCopy()
-	existing.Status = desired.Status
-	src, err := c.RunClientSet.EventsV1alpha1().CloudAuditLogsSources(desired.Namespace).UpdateStatus(existing)
-
-	if err == nil && becomesReady {
-		duration := time.Since(src.ObjectMeta.CreationTimestamp.Time)
-		logging.FromContext(ctx).Desugar().Info("CloudAuditLogsSource became ready", zap.Any("after", duration))
-		c.Recorder.Event(source, corev1.EventTypeNormal, "ReadinessChanged", fmt.Sprintf("CloudAuditLogsSource %q became ready", source.Name))
-		if err := c.StatsReporter.ReportReady("CloudAuditLogsSource", source.Namespace, source.Name, duration); err != nil {
-			logging.FromContext(ctx).Infof("failed to record ready for CloudAuditLogsSource, %v", err)
+func (c *Reconciler) updateStatus(ctx context.Context, existing *v1alpha1.CloudAuditLogsSource, desired *v1alpha1.CloudAuditLogsSource) error {
+	existing = existing.DeepCopy()
+	return reconciler.RetryUpdateConflicts(func(attempts int) (err error) {
+		// The first iteration tries to use the informer's state, subsequent attempts fetch the latest state via API.
+		if attempts > 0 {
+			existing, err = c.auditLogsSourceLister.CloudAuditLogsSources(desired.Namespace).Get(desired.Name)
```

Comment:
here it should be using the client according to the comment, and the other folks code. 
attempt 0 uses the informer copy, which will be that original deepcopied to existing... 

but then if that fails, you query the ApiServer with the clientSet to get a new version, and try to perform the update.. does it make sense?

Reply:
thanks for catching this; error on my part reading the other folk's code. Addressed this one and want to see if tests pass successfully before resolving suggestion you made above.